e2e/settings/settings.po.ts

```diff
@@ -0,0 +1,97 @@
+import {SyndesisComponent} from '../common/common';
+import {$, by, ElementArrayFinder, ElementFinder} from 'protractor';
+import {P} from '../common/world';
+import {log} from '../../src/app/logging';
+
+export class SettingsPage implements SyndesisComponent {
+  rootElement(): ElementFinder {
+    return $('syndesis-settings-root');
+  }
+
+  /**
+   * Get title of active settings tab
+   * @returns {Promise<string>} title of active settings tab
+   */
+  async activeTab(): P<string> {
+    const activeTab = this.rootElement().$('ul.nav-tabs li.active a');
+    return activeTab.getText();
+  }
+
+  /**
+   * Support method for finding settings items by name
+   * @param {ElementFinder} parent search child elements of this element
+   * @returns {ElementArrayFinder} found elements
+   */
+  listSettingsItems(parent: ElementFinder = this.rootElement()): ElementArrayFinder {
+    return parent
+      .$('pfng-list')
+      .$$('div.list-pf-item');
+  }
+
+  /**
+   * Fetch all settings items and find proper one according to given name.
+   * @param {string} name name of settings item
+   * @returns {Promise<ElementFinder | any>} return element or reject promise
+   */
+  async getSettingsItem(name: string): P<ElementFinder | any> {
+    const items = await this.listSettingsItems();
+    log.info(`searching for ${name} in ${items.length} items`);
+
+    for (const item of items) {
+      const title = await item.$('div.list-pf-title').getText();
+      if (title === name) {
+        return item;
+      }
+    }
+    return P.reject(`item ${name} not found`);
+  }
+
+  /**
+   * Click on button which is child element of given settings item
+   * @param {string} settingsItemName name of settings item
+   * @param {string} buttonTitle title of button
+   * @returns {Promise<any>} resolved once clicked
+   */
+  async clickButton(settingsItemName: string, buttonTitle: string): P<any> {
+    const item: ElementFinder = await this.getSettingsItem(settingsItemName);
+    log.info(`click ${buttonTitle} on ${settingsItemName}`);
+    return item.element(by.buttonText(buttonTitle)).click();
+  }
+}
+
+
+/**
+ * Represents oauth apps tab.
+ */
+export class OAuthSettingsComponent extends SettingsPage {
+  rootElement(): ElementFinder {
```

Comment:
I'm not sure inheritance is good approach here.
What do you think about it?

Reply:
I'm not sure to. Oauth apps tab as part of setting page, defined by 'syndesis-settings-root' element, could be better as SettingPage attribute.  But still we could use some kind of general ancestor for tabs.

Reply:
@lvydra can we leave it this way for some time?
I bet we'll have to rewrite it anyway once engineering implements another Tab on this page.